sound/soc/soc-compress.c

```diff
@@ -293,6 +311,14 @@ static int soc_compr_free(struct snd_compr_stream *cstream)
 	}
 
 	mutex_unlock(&rtd->card->pcm_mutex);
+
+	for_each_rtdcom(rtd, rtdcom) {
+		component = rtdcom->component;
+
+		pm_runtime_mark_last_busy(component->dev);
+		pm_runtime_put_autosuspend(component->dev);
+	}
+
```

Comment:
this is the moral equivalent of what ASoC does for PCMs, yes?

Reply:
Precisely. If it bothers too much, I could always add it to intel/probes directly during open and free operations. Again, I feel like the same thing is happening here as it did with allocation. Common code could prove to be a better solution in the long run.

Reply:
that's fine